src/writer.rs

```diff
@@ -15,6 +16,65 @@ pub struct Writer {
     real_n_count: i32
 }
 
+pub trait Writable {
+    fn write(self, w: &mut Writer);
+}
+
+impl Writable for Tractogram {
+    fn write(self, w: &mut Writer) {
+        for item in &self {
+            item.write(w);
+        }
+    }
+}
+
+impl Writable for TractogramItem {
+    fn write(self, w: &mut Writer) {
+        let (streamline, scalars, properties) = self;
+
+        // Transform scalars into a vector of iterators, so we always know where we were
+        let mut scalars = scalars.into_iter().map(|v| v.into_iter()).collect::<Vec<_>>();
```

Comment:
As discussed on Slack, could we link each scalar to it's corresponding points instead of having a `vec![all_red, all_green, all_blue]`? This is how they are written on disk and would probably allow Zipping on points and scalars

```
(p1, s11, s12, s13),
(p2, s21, s22, s23),
...
(pn, sn1, sn2, sn3)
```